### PR TITLE
Add in deployment tools

### PR DESCRIPTION
**Pull request by 0xolishell**
Hey there - great docs! I work for Alchemy and we see that one of the most frequent things developers ask when learning web3 / smart contracts is what to do once they write their first smart contract. Based on these queries, I added in blockchain api providers as well as Goerli faucets to help developers understand how to deploy their solidity based smart contracts onto the Goerli testnet and Ethereum mainnet as I didn't see this in the docs. Love what you guys do for the ecosystem, lmk if there's a better spot to put these :) thanks!

### CODE REVIEW

File: docs/resources.rst [hunk 1]
```diff
@@ -153,3 +153,26 @@ Third-Party Solidity Parsers and Grammars
...
+    
+* `Ankr <https://www.ankr.com/>`_
+
+* `Moralis <https://moralis.io/>`_
+
+* `Quicknode <https://www.quicknode.com/>`_
+
+Goerli Faucet
+=========================================
+Deploying your smart contract to Ethereum mainnet costs real money, aka real ETH. Therefore, it's smart to deploy them first to a testnet. In Ethereum's case, their supported testnet is the Goerli testnet. In order to deploy on the Goerli testnet, you need to first get Goerli testETH (not real money). You get Goerli testETH from a Goerli faucet.
```
Comment (nikola-matic) [inline docs/resources.rst]:
```suggestion
Deploying your smart contract to Ethereum mainnet costs real money, aka real ETH. Therefore, it's smart to deploy them first to a testnet. In Ethereum's case, their supported testnet is the Goerli testnet. In order to deploy on the Goerli testnet, you need to first get Goerli testETH (not real money). You get Goerli testETH from a Goerli faucet.
```

File: docs/resources.rst [hunk 2]
```diff
@@ -153,3 +153,26 @@ Third-Party Solidity Parsers and Grammars
...
+* `Moralis <https://moralis.io/>`_
+
+* `Quicknode <https://www.quicknode.com/>`_
+
+Goerli Faucet
+=========================================
+Deploying your smart contract to Ethereum mainnet costs real money, aka real ETH. Therefore, it's smart to deploy them first to a testnet. In Ethereum's case, their supported testnet is the Goerli testnet. In order to deploy on the Goerli testnet, you need to first get Goerli testETH (not real money). You get Goerli testETH from a Goerli faucet. 
+
+* `goerlifaucet.com <https://goerlifaucet.com>`_
+    Alchemy maintains this free Goerli faucet.
```
Comment (nikola-matic) [inline docs/resources.rst]:
```suggestion
    Alchemy maintains this free Goerli faucet.
```

Comment (0xolishell) [reply docs/resources.rst]: thanks @nikola-matic! Commited the suggestions, appreciate you reviewing

File: docs/resources.rst [hunk 3]
```diff
@@ -153,3 +153,26 @@ Third-Party Solidity Parsers and Grammars
 
 * `Solidity Parser for JavaScript <https://github.com/solidity-parser/parser>`_
     A Solidity parser for JS built on top of a robust ANTLR4 grammar.
+    
+Blockchain API Providers
+=========================================
+When deploying smart contracts to either the Ethereum testnet (Goerli) or mainnet, most developers prefer to use a blockchain developer platform. These platforms provide not only a suite of tooling and abstractions like SDK and NFT APIs, but also, they provide an API endpoint that acts as a wrapper around the underlying JSON RPC API. This API endpoint makes writing to the blockchain easier and faster.
```
Comment (nikola-matic) [inline docs/resources.rst]:
```suggestion
When deploying smart contracts to either the Ethereum testnet (Goerli) or mainnet, most developers prefer to use a blockchain developer platform. These platforms provide not only a suite of tooling and abstractions like SDK and NFT APIs, but also, they provide an API endpoint that acts as a wrapper around the underlying JSON RPC API. This API endpoint makes writing to the blockchain easier and faster.
```

File: docs/resources.rst [hunk 4]
```diff
@@ -153,3 +153,26 @@ Third-Party Solidity Parsers and Grammars
 
 * `Solidity Parser for JavaScript <https://github.com/solidity-parser/parser>`_
     A Solidity parser for JS built on top of a robust ANTLR4 grammar.
+    
+Blockchain API Providers
+=========================================
+When deploying smart contracts to either the Ethereum testnet (Goerli) or mainnet, most developers prefer to use a blockchain developer platform. These platforms provide not only a suite of tooling and abstractions like SDK and NFT APIs, but also, they provide an API endpoint that acts as a wrapper around the underlying JSON RPC API. This API endpoint makes writing to the blockchain easier and faster. 
+
+* `Alchemy <https://alchemy.com>`_
+    A blockchain platform that powers projects like OpenSea, The Graph, Meta, and Shopify.
```
Comment (nikola-matic) [inline docs/resources.rst]:
```suggestion
    A blockchain platform that powers projects like OpenSea, The Graph, Meta, and Shopify.
```

**Review by nikola-matic — COMMENTED** [review]
Some more trailing whitespaces.

File: docs/resources.rst (hunk 3 above)
Comment (0xolishell) [reply docs/resources.rst]: Thanks @nikola-matic, sorry noted for the future to make sure the spacing doesn't appear lie this :)

**Comment by franzihei** [comment]
Hi @shelleyolivia, 

Thanks for contributing and for the kudos!

Unfortunately, the two sections you are suggesting to add are out of scope for the rather lean "Resources" section of the Solidity docs.

I will close this PR due to the following reasons:
- The "Resources" section aims to list resources that are directly relevant to people using Solidity: IDEs, editor integrations and tools. Everything else that is useful or nice to know, like coding tutorials or guides, but covered by other good resources already, is not included in this section. This is especially relevant for topics covered well by the ethereum.org developer portal.
- Blockchain API Providers are well covered by ethereum.org, see the ["Nodes as a service" section](https://ethereum.org/en/developers/docs/nodes-and-clients/nodes-as-a-service/#popular-node-services).
- Testnets and available faucets are well covered by ethereum.org, see the ["Networks" section](https://ethereum.org/en/developers/docs/networks/#ethereum-testnets).

Last but not least, just for the sake of completeness, I'd like to point out that Goerli should not be used as a default testnet for app developers (it's the recommended default testnet for testing of validating and staking). The recommended testnet for application developers is [Sepolia](https://ethereum.org/en/developers/docs/networks/#ethereum-testnets).

**Pull request closed without merging**